### PR TITLE
[PLAYER-1350] VR 360 bugfix 

### PR DESCRIPTION
**Pull request by IvanSakharovskii**

### CODE REVIEW

File: js/components/viewControlsVr.js [hunk 1]
```diff
@@ -1,21 +1,63 @@
...
+var DirectionControlVr = require('./DirectionControlVr');
+var Icon = require('../components/icon');
 var classnames = require('classnames');
+var _ = require('underscore');
 
 var ViewControlsVr = React.createClass({
-  handleDirection: function (rotate, direction) {
-    this.props.controller.moveVrToDirection(rotate, direction);
+  /**
+   *
```
Comment (pilievOoyala) [inline js/components/viewControlsVr.js]: @method tag

Comment (pilievOoyala) [reply js/components/viewControlsVr.js]: and @public

File: js/components/viewControlsVr.js [hunk 2]
```diff
@@ -1,21 +1,63 @@
...
+    if (e.type === "touchend" || !this.props.controller.state.isMobile) {
+      e.stopPropagation(); // W3C
+      e.cancelBubble = true; // IE
+      this.props.controller.state.accessibilityControlsEnabled = true;
+    }
+    this.props.controller.moveVrToDirection(isRotated, direction);
   },
-  
+
+  componentWillMount: function () {
```
Comment (pilievOoyala) [inline js/components/viewControlsVr.js]: JSDocs

File: js/components/viewControlsVr.js [hunk 3]
```diff
@@ -1,21 +1,63 @@
...
+    if (this.props.controller) {
+      if (this.props.controller.videoVrSource) {
+        this.vr = this.props.controller.videoVrSource.vr;
+      }
+
+      if (this.props.controller.state) {
+        this.isMobile = this.props.controller.state.isMobile;
+      }
+
+
```
Comment (pilievOoyala) [inline js/components/viewControlsVr.js]: take out extra lines

File: js/components/viewControlsVr.js [hunk 4]
```diff
@@ -1,21 +1,63 @@
...
   render: function () {
-    return (
-      <div className={classnames("oo-vr-icon-container view-controls", {"oo-vr-icon-container--hidden": !this.props.controlBarVisible})}>
-        <DirectionControlVr {...this.props} handleDirection = {this.handleDirection} dir = "left" />
-        <DirectionControlVr {...this.props} handleDirection = {this.handleDirection} dir = "right" />
-        <DirectionControlVr {...this.props} handleDirection = {this.handleDirection} dir = "up" />
-        <DirectionControlVr {...this.props} handleDirection = {this.handleDirection} dir = "down" />
-      </div>
-    );
+    var isShow = !!(this.icon && this.icon.name);
```
Comment (pilievOoyala) [inline js/components/viewControlsVr.js]: toShow or isShowing

File: js/skin.js [hunk 5]
```diff
@@ -68,6 +72,116 @@ var Skin = React.createClass({
     });
   },
 
+  /**
+   * @description the function is called when we start the rotation
+   * @param e - event
```
Comment (pilievOoyala) [inline js/skin.js]: @public or @private tags for all these functions.

File: js/skin.js [hunk 6]
```diff
@@ -68,6 +72,116 @@ var Skin = React.createClass({
...
+        this.props.controller.onTouchMove(params, true);
+      }
+    }
+  },
+
+  /**
+   * @description the function is called when we stop the rotation
+   */
+  handleVrPlayerMouseUp: function() {
+    if (this.props.controller.videoVr) {
```
Comment (pilievOoyala) [inline js/skin.js]: this.props.controller is guaranteed to exist at this point, correct?

File: js/components/viewControlsVr.js [hunk 7]
```diff
@@ -1,21 +1,63 @@
...
+        this.vr = this.props.controller.videoVrSource.vr;
+      }
+
+      if (this.props.controller.state) {
+        this.isMobile = this.props.controller.state.isMobile;
+      }
+
+
+      if (!this.props.controller.state.isPlayingAd) {
+        if (!this.isMobile) {
```
Comment (aldigun) [inline js/components/viewControlsVr.js]: These 2 IFs can be combined into a single line.

File: scss/components/_state-screen.scss [hunk 8]
```diff
@@ -30,6 +30,7 @@
     @extend .oo-expand;
     text-decoration:none;
     cursor: default;
+    z-index: 1000;
```
Comment (pilievOoyala) [inline scss/components/_state-screen.scss]: Do we have a place where we define constants for Z-indexes?  If so can we use those.

File: js/skin.js [hunk 9]
```diff
@@ -68,6 +72,116 @@ var Skin = React.createClass({
...
+  /**
+   * @description get direction params. Direction params are values for new position of a vr video (yaw, roll=0, pitch)
+   * @private
+   * @param pageX {number} x coordinate
+   * @param pageY {number} y coordinate
+   * @returns {[number, number, number]}
+   */
+
+  getDirectionParams: function(pageX, pageY) {
+    var dx = pageX - this.state.xVrMouseStart;
```
Comment (aldigun) [inline js/skin.js]: what if pageX and pageY are not a type of number?

**Comment by pilievOoyala** [comment]
LGTM

File: js/views/pauseScreen.js [hunk 10]
```diff
@@ -107,22 +127,49 @@ var PauseScreen = React.createClass({
 
     var titleMetadata = (<div className={titleClass} style={titleStyle}>{this.props.contentTree.title}</div>);
     var descriptionMetadata = (<div className={descriptionClass} ref="description" style={descriptionStyle}>{this.state.descriptionText}</div>);
+
```
Comment (aldigun) [inline js/views/pauseScreen.js]: remove extra lines

**Comment by jpfranco** [comment]
@IvanSakharovskii please be careful not to include local changes in pull requests. I'm seeing that the ScrollArea component was removed from the code, possibly to fix the React 16 build error. The workaround for the build issue is to use `npm dedupe` which has been added to the readme file, so please do not commit local environment workarounds to your feature branches.

**Pull request merged**